Python/ast.c

```diff
@@ -1686,19 +1685,14 @@ ast_for_decorated(struct compiling *c, const node *n)
            TYPE(CHILD(n, 1)) == classdef);
 
     if (TYPE(CHILD(n, 1)) == funcdef) {
-      thing = ast_for_funcdef(c, CHILD(n, 1), decorator_seq);
+        return ast_for_funcdef(c, CHILD(n, 1), decorator_seq);
     } else if (TYPE(CHILD(n, 1)) == classdef) {
```

Comment:
Actually now you can remove all `else`s.

Reply:
Even with `Py_UNREACHABLE()`?

Reply:
No, with the `Py_UNREACHABLE()` it is happy. Do you want something like this?
```
    if (TYPE(CHILD(n, 1)) == funcdef) {
        return ast_for_funcdef(c, CHILD(n, 1), decorator_seq);
    } else if (TYPE(CHILD(n, 1)) == classdef) {
        return ast_for_classdef(c, CHILD(n, 1), decorator_seq);
    } else if (TYPE(CHILD(n, 1)) == async_funcdef) {
        return ast_for_async_funcdef(c, CHILD(n, 1), decorator_seq);
    }
    Py_UNREACHABLE();
```

Reply:
I suggest to remove other `else`s too.